### DISCUSSION TITLE
Caret is jumping when I call Update function.

### DISCUSSION THREAD

**Discussion by VilikNakhshikyan** [phosphor-icons/homepage#551] [General]
I have a widget that inherits from ReactDialog from theia and ReactDialog inherits from Widget (from phosphor) it has an update function.
For example, I have a search field and with each type I call this.update() as a result of which the cursor jumps to the end of the line.
Is there any solution?
![Screenshot from 2024-08-27 11-10-23](https://github.com/user-attachments/assets/c1707ea4-df31-475a-9974-62066e68cd58)

https://github.com/user-attachments/assets/2a908edf-f258-4df2-ab8a-f57730c7dcaf

**Comment by rektdeckard** [comment]
I think you have us confused with [@phosphor/widgets](https://www.npmjs.com/package/@phosphor/widgets), a (now abandoned) UI framework. Your issue is unrelated to Phosphor Icons.
